awscli/clidocs.py

```diff
@@ -304,7 +304,21 @@ def build_translation_map(self):
         d = {}
         for cli_name, cli_argument in self.help_command.arg_table.items():
             if cli_argument.argument_model is not None:
-                d[cli_argument.argument_model.name] = cli_name
+                argument_name = cli_argument.argument_model.name
+                if argument_name in d:
+                    previous_mapping = d[argument_name]
+                    # If the argument name is a boolean argument, we want the
+                    # the translation to default to the one that does not start
+                    # with --no-. So we check if the cli parameter currently
+                    # being used starts with no- and if stripping off the no-
+                    # results in the new proposed cli argument name. If it
+                    # does, we assume we have the postive form of the argument
+                    # which is the name we want to use in doc translations.
+                    if previous_mapping.startswith('no-') and \
```

Comment:
This seems like it has potential for false positives.  Can we restrict this check to be more specific?  What about checking if arg is an instance of the boolean argument?